### PR TITLE
Baseproject

### PR DESCRIPTION
**Pull request by joelgtsantos**
Rest project contains all the entities to register a new user into the db

### CODE REVIEW

File: src/main/java/com/cms/users/entity/UserBase.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	 *Using as main key the string in the cms-config file
+	 * 
+	 * */
+	@PrePersist
+	public void userPrePersist(User user) {
+		String newPassword = "";
+		
+		try {
+		String input = user.getPassword();
+		String key = "8e045a51e4b102ea803c06f92841a1fb";
```
Comment (jossemarGT) [inline src/main/java/com/cms/users/entity/UserBase.java:28]: This has to be a configurable property instead of harcoding it in the entity

Comment (jossemarGT) [reply src/main/java/com/cms/users/entity/UserBase.java:28]: The same goes for the next line

File: src/main/java/com/cms/users/config/filters/CORSFilter.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+	@Override
+	public void init(FilterConfig filterConfig) throws ServletException {
+	}
+
+	@Override
+	public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain filterChain)
+			throws IOException, ServletException {
+		HttpServletResponse response = (HttpServletResponse) servletResponse;
+		HttpServletRequest request = (HttpServletRequest) servletRequest;
+		response.addHeader("Access-Control-Allow-Methods", "GET,POST,PUT,DELETE,PATCH");
```
Comment (jossemarGT) [inline src/main/java/com/cms/users/config/filters/CORSFilter.java:26]: I'm not sure of letting the `DELETE` verb for the current iteration

File: src/main/java/com/cms/users/entity/UserBase.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+		String initVector = "RandomInitVector";//Random Key
+		
+		 IvParameterSpec iv = new IvParameterSpec(initVector.getBytes("UTF-8"));
+		 SecretKeySpec skeySpec = new SecretKeySpec(key.getBytes("UTF-8"), "AES");
+		
+		 Cipher cipher = Cipher.getInstance("AES/CBC/PKCS5PADDING");
```
Comment (jossemarGT) [inline src/main/java/com/cms/users/entity/UserBase.java:34]: I'm curious, are we storing the CMS users password as plaintext or as bcript value?

File: src/main/resources/application.yml [hunk 4]
```diff
@@ -0,0 +1,45 @@
+server:
```
Comment (jossemarGT) [inline src/main/resources/application.yml:1]: As I mentioned in `src/main/java/com/cms/users/entity/UserBase.java`, I would spec to see the secrets here or just a placeholder for environment variables

File: src/main/java/com/cms/users/entity/UserBase.java (hunk 3 above)
Comment (joelgtsantos) [reply src/main/java/com/cms/users/entity/UserBase.java:34]: Right now I'm working with plain/text but it needs to be AES iv

File: src/main/resources/application.yml (hunk 4 above)
Comment (joelgtsantos) [reply src/main/resources/application.yml:1]: Thank you for this observation next release will include this property

**Pull request merged**